docs: Fix inaccuracies, unify wording, and expand SDK documentation (#921)
apify/apify-sdk-python: docs/02_concepts/08_access_apify_api.mdx

The Apify SDK provides a built-in instance of the [Apify API Client](https://docs.apify.com/api/client/python) for accessing Apify platform features beyond what the SDK covers directly.

> The SDK wraps the most common operations (opening storages, reading input, charging, and starting other Actors) in high-level methods. When you need something it does not expose directly, such as managing schedules, listing your Actors and runs, working with builds or webhook dispatches, or reading another run's dataset, you can reach the full [Apify API](https://docs.apify.com/api/v2) through the client. It provides a dedicated sub-client for each API resource (for example `dataset`, `run`, or `schedule`), and each operation returns plain Python data structures.

Reviewer: ```suggestion
The SDK wraps the most common operations (opening storages, reading input, charging, and starting other Actors) in high-level methods. To do things that it doesn't expose directly, such as managing schedules, listing your Actors and runs, working with builds or webhook dispatches, or reading another run's dataset, you can reach the full [Apify API](https://docs.apify.com/api/v2) through the client. It provides a dedicated sub-client for each API resource (for example `dataset`, `run`, or `schedule`), and each operation returns plain Python data structures.
```

## Actor client

To access the provided instance of [`ApifyClientAsync`](https://docs.apify.com/api/client/python/reference/class/ApifyClientAsync), you can use the <ApiLink to="class/Actor#apify_client">`Actor.apify_client`</ApiLink> property. This instance is already authenticated with the Actor's API token and points at the same API the Actor runs against.

For example, to get the details of your user, you can use this snippet:

<RunnableCodeBlock className="language-python" language="python">
    {ActorClientExample}
</RunnableCodeBlock>

You select a resource with the matching method (for example, `Actor.apify_client.dataset('<dataset_id>')` for a dataset or `Actor.apify_client.run('<run_id>')` for an Actor run), and then call the operation you need on the returned sub-client.

## Actor new client

> If you want to create a completely new instance of the client (for example, to use a different user's token or change the client configuration), you can use the <ApiLink to="class/Actor#new_client">`Actor.new_client`</ApiLink> method. It accepts an optional `token` and `api_url`, along with retry and timeout settings (`max_retries`, `min_delay_between_retries`, `timeout`), and returns a fresh `ApifyClientAsync`:

Reviewer: ```suggestion
To create a completely new instance of the client (for example, to use a different user's token or change the client configuration), use the <ApiLink to="class/Actor#new_client">`Actor.new_client`</ApiLink> method. It accepts an optional `token` and `api_url`, along with retry and timeout settings (`max_retries`, `min_delay_between_retries`, `timeout`), and returns a fresh `ApifyClientAsync`:
```
